enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,603 @@
+2. As a user/devops/developer, I need to configure Route to set/append HTTP Response headers which need to  which 
+   are mentioned above to send them to the client which has called the route.
+3. To satisfy this use-case, the cluster administrator can set the IngressController's Spec 
+   or the developer can set Router's `spec.httpDeleteHeaders` field to set(replace) or append the required headers.
+4. As a cluster administrator, I need to configure an IngressController to delete certain HTTP Response headers which are:
+   - X-XSS-Protection
+   - X-Content-Type-Options
+   - X-Frame-Options
+     in order to satisfy compliance requirements at global level to not to send them to all clients.
+5. As a user/devops/developer, I need to configure Route to delete HTTP Response headers which need to  which
+      are mentioned above for not sending them to the client which has called the route.
+6. To satisfy this use-case, the cluster administrator can set the IngressController's Spec
+   or the developer can set the Router's Spec with `spec.httpDeleteHeaders` field to delete the required headers by 
+   mentioning the header names mentioned above.
+
+### Goals
+
+1. Enable the cluster administrator to specify HTTP response headers in the IngressController CR spec that should get set or appended at the global
+   level, i.e. for all routes.
+2. Enable the route owner to specify HTTP response headers in the Route spec (when it is not a passthrough Route) for their application
+   that should get set or appended when forwarding a response to the client.
+3. Enable the cluster administrator to delete the HTTP response headers via IngressController CR.
+4. Enable the developer to delete the HTTP response headers via RouteSpec.
+5. This EP will set an immutable/hardcoded deny list of headers not allowed to set/append or deleted.
+   Eg- we should prohibit specifying strict-transport-security using the new API, for the following reasons:
+   HSTS is already configurable through a route annotation.
+   We already have an API for restricting HSTS in the cluster config.
+   Adding a second way to configure HSTS would significantly complicate this enhancement.
+
+### Non-Goals
+
+1. Validation that each specified value is valid for the respective header (e.g. require that either "DENY" or "SAMEORIGIN" is 
+   specified for [the `x-frame-options` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options)).
+2. Setting load balancing algorithm using header values.
+   This request is orthogonal to customizing HTTP headers. It is out of scope for this enhancement and should be handled
+   with a seprate RFE.
+3. Configuring buffer sizes. Eg- tune.bufsize and tune.maxrewrite
+4. Modifying headers for a passthrough route.
+
+## Proposal
+
+### Workflow Description
+
+**Cluster admin** is a human user responsible for deploying a
+cluster.
+
+**Users/Devops/Developers** is a human user responsible for developing and
```

Comment:
I think that the end user of the API is a different actor from the developer which designs the API. The difference between them can be seen, for instance, in the part where the deny list is defined. The user actor cannot influence this list in any form, while developer has to be define it.
Also what did you mean by the devops actor? How it's different from the user of the API?

Reply:
I meant application developer

Reply:
I see, then maybe this can be condensed. `User/DevOps/Application Developer` may simply become `User`. And I think that `Developer` as an actor representing NetworkEdge Engineer is still needed, for the deny list use case (user story and other mentions).

Reply:
changed